external/java/src/test/java/ch/epfl/dedis/lib/omniledger/contracts/EventLogInstanceTest.java

```diff
@@ -70,11 +71,24 @@ void testLogMore() throws Exception {
             // add all the events. So we have to limit the number of events that we add.
             keys.add(el.log(event, Arrays.asList(admin)));
         }
-        Thread.sleep(2 * ol.getConfig().getBlockInterval().toMillis());
-        for (InstanceId key : keys) {
-            Event event2 = el.get(key);
-            assertEquals(event, event2);
+        boolean allOK = true;
+        for (int i = 0; i < 4; i++) {
+            allOK = true;
+            Thread.sleep(2 * ol.getConfig().getBlockInterval().toMillis());
+            for (InstanceId key : keys) {
+                try {
+                    Event event2 = el.get(key);
+                    assertEquals(event, event2);
+                } catch (CothorityCryptoException e){
```

Comment:
why don't we just let the exception be thrown? the test will fail and i think that achieves the same effect.

Reply:
Because we try four times to get all the proofs - so if we don't succeed in one of the first three rounds, we don't want the exception to be thrown.